### PR TITLE
refactor: 💡 collapse state stored in provenance and grammar

### LINKED ISSUE #139: Add Collapsed flags to grammar
Add the collapsed state object to Trrack

### PR DESCRIPTION
**Pull request by JakeWags**
### Does this PR close any open issues?
Closes #139 

### Give a longer description of what this PR addresses and why it's needed
This PR adds the collapsed ids to the provenance and the grammar as a list of strings. Collapse all appends all ids to the array in a single provenance action. Similarly, expand all removes all items in a single action.

### Provide pictures/videos of the behavior before and after these changes (optional)
Provenance example:
image

Exported grammar from the state pictured above: 
```
{
  "firstAggregateBy": "Degree",
  "firstOverlapDegree": 2,
  "secondAggregateBy": "Sets",
  "secondOverlapDegree": 2,
  "sortBy": "Cardinality",
  "filters": {
    "maxVisible": 3,
    "minVisible": 0,
    "hideEmpty": true
  },
  "visibleSets": [
    "Set_Action",
    "Set_Adventure",
    "Set_Children",
    "Set_Comedy",
    "Set_Crime",
    "Set_Documentary"
  ],
  "visibleAttributes": [
    "ReleaseDate",
    "AvgRating",
    "Watches"
  ],
  "bookmarkedIntersections": [],
  "collapsed": [
    "Agg_Degree_0",
    "Agg_Degree_1",
    "Agg_Degree_1-Agg_Comedy",
    "Agg_Degree_2-Agg_Action",
    "Agg_Degree_2-Agg_Children",
    "Agg_Degree_3-Agg_Crime",
    "Agg_Degree_4"
  ],
  "plots": {
    "scatterplots": [],
    "histograms": [],
    "wordClouds": []
  }
}
```

### Are there any additional TODOs before this PR is ready to go?

TODOs:
- [X] Local testing

### CODE REVIEW

**Comment by JakeWags** [comment]
@filemon-mateus Note the change in grammar to include "collapsed". This is an array of strings which represent the aggregates which have been collapsed. If the aggregate is not in the array, then it is **not** collapsed.

File: packages/upset/src/components/Header/CollapseAllButton.tsx [hunk 1]
```diff
@@ -20,21 +22,31 @@ const collapseAllStyle = css`
...
-      })
-  
-      setAllCollapsed(collapsed);
-      setCollapsedIds(ids);
+        const collapsed = !allCollapsed;
+        const ids: string[] = [];
+
+        if (allCollapsed === true) {
+            actions.expandAll();
+            setAllCollapsed(collapsed);
```
Comment (JackWilb) [inline packages/upset/src/components/Header/CollapseAllButton.tsx:37]: I think it's a bit hard to follow since you're using collapsed here, and `collapsed = !allCollapsed`. Can we just use the values true and false, and not this variable?

File: packages/upset/src/components/Header/CollapseAllButton.tsx [hunk 2]
```diff
@@ -20,21 +22,31 @@ const collapseAllStyle = css`
...
+            return;
```
Comment (JackWilb) [inline packages/upset/src/components/Header/CollapseAllButton.tsx:38]: This return means we don't do if else, but it would be more explicit if we remove this return and do if else, instead

**Review by JackWilb — APPROVED** [review]
Looks good. I had a couple of suggestions for increasing the readability and maintainability of some of the additions
